src/apps/lwaftr/podhashmap.lua

```diff
@@ -36,11 +36,34 @@ local function hash_to_index(hash, scale)
    return floor(hash*scale + 0.5)
 end
 
+local function make_equal_fn(key_type)
+   local size = ffi.sizeof(key_type)
+   if pcall(tonumber, ffi.new(key_type)) then
+      return function (a, b)
+         return a == b
+      end
+   elseif size == 4 then
+      local cast = ffi.cast
+      return function (a, b)
+         return cast('uint32_t[1]', a)[0] == cast('uint32_t[1]', b)[0]
+      end
+   elseif size == 8 then
+      return function (a, b)
+         return cast('uint64_t[1]', a)[0] == cast('uint64_t[1]', b)[0]
```

Comment:
`cast` is localized in the `if size == 4 then` branch but not here. Is this correct?